sdk/core/azure-core-test/src/main/java/com/azure/core/test/utils/TestProxyUtils.java

```diff
@@ -320,51 +327,112 @@ private static String createRegexRequestBody(String key, String regex, String va
+     * <p>
+     * For more information about adding bulk sanitizers see the
+     * <a href="https://github.com/Azure/azure-sdk-tools/tree/feature/add-bulk-sanitizers/tools/test-proxy/Azure.Sdk.Tools.TestProxy#passing-sanitizers-in-bulk">Passing Sanitizers in Bulk</a>
+     * wiki.
+     *
+     * @param sanitizers The list of sanitizers to be added.
+     * @param proxyUrl The proxyUrl to use when constructing requests.
+     * @return The {@link HttpRequest request} to be sent.
+     * @throws RuntimeException if {@link TestProxySanitizerType} is not supported.
+     */
+    public static HttpRequest createAddSanitizersRequest(List<TestProxySanitizer> sanitizers, URL proxyUrl) {
+        List<String> sanitizersJsonPayloads = new ArrayList<>(sanitizers.size());
+
+        for (TestProxySanitizer sanitizer : sanitizers) {
+            String requestBody;
+            String sanitizerType;
+            switch (sanitizer.getType()) {
+                case URL:
+                    sanitizerType = TestProxySanitizerType.URL.getName();
+                    requestBody = createRegexRequestBody(null, sanitizer.getRegex(), sanitizer.getRedactedValue(),
+                        sanitizer.getGroupForReplace());
+                    break;
+
+                case BODY_REGEX:
+                    sanitizerType = TestProxySanitizerType.BODY_REGEX.getName();
+                    requestBody = createRegexRequestBody(null, sanitizer.getRegex(), sanitizer.getRedactedValue(),
+                        sanitizer.getGroupForReplace());
+                    break;
+
+                case BODY_KEY:
+                    sanitizerType = TestProxySanitizerType.BODY_KEY.getName();
+                    requestBody = createBodyJsonKeyRequestBody(sanitizer.getKey(), sanitizer.getRegex(),
+                        sanitizer.getRedactedValue());
+                    break;
+
+                case HEADER:
+                    sanitizerType = HEADER.getName();
+                    if (sanitizer.getKey() == null && sanitizer.getRegex() == null) {
+                        throw new RuntimeException(
+                            "Missing regexKey and/or headerKey for sanitizer type {" + sanitizerType + "}");
+                    }
+                    requestBody = createRegexRequestBody(sanitizer.getKey(), sanitizer.getRegex(), sanitizer.getRedactedValue(),
+                        sanitizer.getGroupForReplace());
+                    break;
+
+                default:
+                    throw new RuntimeException("Sanitizer type {" + sanitizer.getType() + "} not supported");
+            }
+
+            sanitizersJsonPayloads.add("{\"Name\":\"" + sanitizerType + "\",\"Body\":" + requestBody + "}");
+        }
+
+        String requestBody = "[" + CoreUtils.stringJoin(",", sanitizersJsonPayloads) + "]";
+        return new HttpRequest(HttpMethod.POST, proxyUrl +"/Admin/AddSanitizers").setBody(requestBody);
```

Comment:
I was also thinking to make this API public so that SDK's like resource manager can add sanitizers in bulk.

Reply:
This API is public 😄

Reply:
Oh I meant something on the interceptor manager that SDK's can use too to add bukl sanitizers. I can add it in a PR later.